addons/mrp/static/src/widgets/timer.js

```diff
@@ -94,6 +95,9 @@ class MrpTimerField extends Component {
             parse: (v) => parseFloatTime(v),
         });
 
+        // duration is expected to be given in minutes
+        this.duration = this.props.record.data[this.props.name];
+
```

Comment:
The definition of `this.duration` in the `useRecordObserver` is enough, this one and the `onWillStart` can be removed I think
```suggestion
```